synfig-studio/src/gui/widgets/widget_timeslider.cpp

```diff
@@ -349,21 +388,65 @@ bool
 Widget_Timeslider::on_button_release_event(GdkEventButton *event){
 	SYNFIG_EXCEPTION_GUARD_BEGIN()
 	lastx = (double)event->x;
+	move_lower_bound_button = false;
+	move_upper_bound_button = false;
 	return event->button == 1 || event->button == 2;
 	SYNFIG_EXCEPTION_GUARD_END_BOOL(true)
 }
 
+void
+Widget_Timeslider::get_bounds_rectangle_dimensions(double& x0, double& x1, double& w ,bool lower)
+{
+	if(lower){
+		 x0 = time_plot_data->get_double_pixel_t_coord(time_plot_data->lower_ex);
+		 x1 = time_plot_data->get_double_pixel_t_coord(time_plot_data->time_model->get_play_bounds_lower());
+		 w = x1 - x0;
+	}
+	else {
+		 x0= time_plot_data->get_double_pixel_t_coord(time_plot_data->time_model->get_play_bounds_upper());
+		 x1= time_plot_data->get_double_pixel_t_coord(time_plot_data->upper_ex);
+		 w= x1 - x0;
+	}
+}
+
 bool
 Widget_Timeslider::on_motion_notify_event(GdkEventMotion* event) //for dragging
 {
 	SYNFIG_EXCEPTION_GUARD_BEGIN()
+
+	double current= time_plot_data->get_double_pixel_t_coord(time_plot_data->get_t_from_pixel_coord(event->x));
```

Comment:
`current` what? :) Why this conversion followed by reversed conversion?

Reply:
hmm looks like I overcomplicated this for no reason :" I thought the "pixel coordinates" where a bit diffrent from the events coordinate but turns out they are the same.